// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - LivingEdgeLab
//
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";
package protocol.flight_log_service;

import "common.proto";

/*
 * Used to log to a flight log
 */
service FlightLog {
  // Basic log endpoint 
  rpc Log (LogRequest) returns (LogResponse) {}
}

message LogRequest {
  protocol.common.Request request = 1;
  string topic = 2;
  LogMessage log = 3;
}

message LogResponse {
  protocol.common.Response response = 1;
}

enum LogType {
  INFO = 0;
  DEBUG = 1;
  WARNING = 2;
  ERROR = 3;
}

message LogMessage {
  LogType type = 1;
  string msg = 2;
}
